ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/Util/AcidState.hs

```diff
@@ -101,3 +108,69 @@ zoomAll l upd = StateT $ \large -> do
 mustBeRight :: Either Void b -> b
 mustBeRight (Left  a) = absurd a
 mustBeRight (Right b) = b
+
+type MonadAcidCleanup m =
+    ( MonadIO m
+    , MonadMask m
+    )
+
+
+-- | Helper function to run action periodically.
+runPeriodically :: forall (unit :: Rat) m a. (KnownDivRat unit Second, MonadIO m)
+  => Time unit -- ^ time between performing action
+  -> m a       -- ^ action
+  -> m ()
+runPeriodically delay action = forever $ do
+  _ <- action
+  threadDelay delay
+
+
+-- | Creates checkpoint of the current state of Acid DB,
+-- archive previous checkpoints and remove too old archives.
+-- This is needed to avoid storing all logs permanently.
+cleanupAcidState ::
+       forall m st. (MonadAcidCleanup m)
+    => AcidState st
+    -> FilePath
+    -> Int
+    -> (Severity -> Text -> m ())
+    -> m ()
+cleanupAcidState db path numberOfStoredArchives writeLog = perform
+  where
+    perform = cleanupAction `catchAny` handler
+
+    cleanupAction = do
+        writeLog Debug "Starting cleanup"
+        -- checkpoint/archive
+        liftIO $ createCheckpoint db >> createArchive db
```

Comment:
It is not clear to me from the [docs](https://hackage.haskell.org/package/acid-state-0.14.3/docs/Data-Acid.html#v:createArchive) that `createArchive` will not return until the operation has succeeded. Do you have any evidence to support that?

Reply:
I read sources and can't find any async IO operations. 
The last function ["archiveFileLog"](https://github.com/acid-state/acid-state/blob/2952267f23ca0ee757ff724c7911f3034eb8c7af/src/Data/Acid/Log.hs#L260) uses standard functions from "System.Directory". I don't think that "renameFile" can return until the operation has succeeded.